### ISSUE TITLE
[Bug]: Using xycoords=("data","subfigure fraction") or ("data", subfigure points") breaks annotation position when exporting

### ISSUE THREAD

**Issue by rk-exxec** [matplotlib/matplotlib#27438]
### Bug summary

When adjusting the annotation position within the "subfigure fraction" or "subfigure points" coordinate system, the results differ between the Jupyter inline preview and exported files. (Probably also with "subfigure pixels", but I didn't test.)
In my case, the annotations are located outside of the axes region.

### Code for reproduction

```python
import numpy as np
import matplotlib.pyplot as plt

x = np.arange(0, 5*np.pi, 0.1)
y = np.sin(x)

fig = plt.figure(dpi=300)
axes = fig.subplots(1,1,sharex=True)
axes.plot(x, y)

axes.annotate("Fraction", xy=(3, 0.93), xycoords=("data","subfigure fraction"), annotation_clip=False)
axes.annotate("Points", xy=(6, 297), xycoords=("data","subfigure points"), annotation_clip=False)
plt.tight_layout()

fig.savefig("png_image.png", dpi=300)
fig.savefig("pdf_image.pdf", dpi=300)
```

### Actual outcome

The "Fraction" text moves upwards while the "Points" text moves downwards in comparison to the Jupyter Preview below. In both PNG exports and PDF exports.

![bug](https://github.com/matplotlib/matplotlib/assets/40037381/e574c410-850b-477e-887e-1fe51a1b6786)

### Expected outcome

The two annotations should be at a very similar y position.

![grafik](https://github.com/matplotlib/matplotlib/assets/40037381/0b22958b-e8a4-44ba-a993-8036c56061f8)

### Additional information

This can be circumvented when using both `plt.tight_layout()` for the inline plot and `bbox_inches="tight"` during export.   
But this should just work no?
This implies that the images are somehow different size even though the dpi is the same.

### Operating system

Windows 10 

### Matplotlib Version

3.8.2

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

3.11.4

### Jupyter version

6.5.4

### Installation

pip

**Comment by rcomer** [comment]
`bbox_inches="tight"` does indeed change the size of the figure so that it fits the artists on it.  The inline backend automatically uses `bbox_inches="tight"`

**Comment by jklymak** [comment]
I don't think this is a bug per-se, nor is there much we can do about it.  Please see https://stackoverflow.com/a/37879281/3394386 for how to disable the default in the inline backend.

**Comment by rcomer** [comment]
I’m going to close this because, as @jklymak notes, it isn't a bug in Matplotlib.  You are welcome to open a question over at https://discourse.matplotlib.org/ if you need more help.

**Comment by rk-exxec** [comment]
Ok, thank you for the workaround @jklymak.
